nimdoc/testproject/expected/subdir/subdir_b/utils.html

```diff
@@ -20,7 +20,7 @@
 <title>subdir/subdir_b/utils</title>
 <link rel="stylesheet" type="text/css" href="../../nimdoc.out.css">
 
-<script type="text/javascript" src="dochack.js"></script>
+<script type="text/javascript" src="../../dochack.js"></script>
```

Comment:
Are you certain this is correct?

Reply:
yes, it's a bugfix i didn't bother to mention, feel free to try this PR with instructions I gave in updated `doc/docgen.rst`: because of CORS, `python3 -m http.server 7029 --directory htmldocs` is needed for testing out search feature.

```
nim doc --project --outdir:/tmp/d58 --index nimdoc/testproject/testproject.nim

tree /tmp/d58
/tmp/d58
├── dochack.js
├── nimdoc.out.css
├── subdir
│   └── subdir_b
│       ├── utils.html
│       └── utils.idx
├── testproject.html
├── testproject.idx
└── theindex.html
```

as noted in top post, dochack.js href is correct but the hardcoded `request.open("GET", "theindex.html", false);` is a pre-existing bug left as TODO for a future PR after this is merged

## before this PR:
you'd either need to manually copy dochack.js in several locations, or get an invalid href

##  after this PR:
dochack.js is automatically compiled from dochack.nim if that file doesn't yet exist, and then dochack.js is then copied to root of `$outdir`, and all hrefs to it should be correct